### ISSUE TITLE
BYONM: Dynamically importing CommonJS file from an npm package errors

### ISSUE THREAD

**Issue by bartlomieju** [denoland/deno#21149]
```
$ deno run -A npm:create-next-app@latest next-app
$ cd next-app
$ deno run -A --unstable-byonm npm:next dev
error: Uncaught (in promise) ReferenceError: module is not defined
module.exports = nextConfig
^
    at file:///Users/ib/dev/next-app/next.config.js:6:1
```

Chaging `module.exports =` to `export default` fixes the problem.

**Comment by Cedar-81** [comment]
is this a problem that needs to be fixed on Denos end? (Still new to open source, just trying to understand how it could be fixed)

**Comment by dsherret** [comment]
I disagree that this is a bug. The file is not located in an npm package. It's located at `file:///Users/ib/dev/next-app/next.config.js`, which means it is ESM. This can be fixed in next.config.js by changing:

```
module.exports = nextConfig;
```

To...

```
export default nextConfig;
```

...as Next.js loads this file via a dynamic import. We should update the error message to explain that `module.exports =` needs to be converted to ESM.

**Comment by bartlomieju** [comment]
I'd say this is fixed. If one adds `type: "commonjs"` to `package.json` then it works fine.
